crates/wasmtime/src/engine.rs

```diff
@@ -69,39 +39,26 @@ struct EngineInner {
     compiler: Compiler,
     allocator: Box<dyn InstanceAllocator>,
     signatures: SignatureRegistry,
-    host_funcs: EngineHostFuncs,
 }
 
 impl Engine {
     /// Creates a new [`Engine`] with the specified compilation and
     /// configuration settings.
     pub fn new(config: &Config) -> Result<Engine> {
+        // Ensure that wasmtime_runtime's signal handlers are configured. This
+        // is the per-program initialization required for handling traps, such
+        // as configuring signals, vectored exception handlers, etc.
+        wasmtime_runtime::init_traps(crate::module::GlobalModuleRegistry::is_wasm_pc);
```

Comment:
I moved this here from `Store`, mostly just for aesthetic reasons, no real motivation other than that. This could arguably move to `Config` too, but it felt a little more appropriate here. Doesn't really matter one way or another though.